### PR TITLE
Properties storing in AWS parameter store

### PR DESCRIPTION
**Pull request by Arundhathy-k**
Properties from application.properties file is stored into aws parameter store and database connected manually using those properties from parameter store.

### CODE REVIEW

File: contentService/src/main/java/com/kovan/ContentServiceMain.java [hunk 1]
```diff
@@ -2,9 +2,11 @@
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration;
 
-@SpringBootApplication
+@SpringBootApplication(exclude = DataSourceAutoConfiguration.class)
```
Comment (dhanyesh) [inline contentService/src/main/java/com/kovan/ContentServiceMain.java]: why you added exclude DataSourceAutoConfiguration class ?

File: contentService/src/main/resources/application.properties [hunk 2]
```diff
@@ -1,10 +1,12 @@
...
-spring.datasource.username=root
-spring.datasource.password=12345
-spring.jpa.properties.hibernate.dialect=org.hibernate.dialect.MySQL8Dialect
-spring.jpa.hibernate.ddl-auto=update
-spring.jpa.properties.hibernate.show_sql=true
-spring.jpa.properties.hibernate.format_sql=true
\ No newline at end of file
+spring.config.import=aws-parameterstore:/config/contentService/
+
+#spring.datasource.url=${spring.datasource.url}
```
Comment (dhanyesh) [inline contentService/src/main/resources/application.properties]: If you don't want, please remove this don't use comment here.

File: mongo-dao/src/main/resourses/application.properties [hunk 3]
```diff
@@ -1,2 +1,2 @@
-spring.data.mongodb.uri= mongodb+srv://Arundhathy:Ammu2002@kovan.us1gd.mongodb.net/?retryWrites=true&w=majority&appName=Kovan
-spring.data.mongodb.database = NewsDB
+spring.config.import=aws-parameterstore:/config/NewsService/
+spring.data.mongodb.uri= ${MongoDbURI}
```
Comment (dhanyesh) [inline mongo-dao/src/main/resourses/application.properties]: Is this spring.data.mongodb.uri= ${MongoDbURI} needed ?

File: newsService/src/main/resources/application.properties [hunk 4]
```diff
@@ -1,10 +1,6 @@
...
-spring.data.mongodb.uri= mongodb+srv://Arundhathy:Ammu2002@kovan.us1gd.mongodb.net/?retryWrites=true&w=majority&appName=Kovan
-spring.data.mongodb.database = NewsDB
 
-news.api.url=https://newsapi.org/v2/top-headlines
-news.api.country=us
-news.api.category=technology
-news.api.key=71a2d5b0f83f460b890e3202b1f1fc55
\ No newline at end of file
+spring.config.import=aws-parameterstore:/config/NewsService/
+spring.data.mongodb.uri= ${MongoDbURI}
```
Comment (dhanyesh) [inline newsService/src/main/resources/application.properties:6]: Do we need this spring.data.mongodb.uri= ${MongoDbURI}

**Review by dhanyesh — CHANGES_REQUESTED** [review]

File: contentService/src/main/resources/application.properties (hunk 2 above)
Comment (Arundhathy-k) [reply contentService/src/main/resources/application.properties]: Removed

File: contentService/src/main/java/com/kovan/ContentServiceMain.java (hunk 1 above)
Comment (Arundhathy-k) [reply contentService/src/main/java/com/kovan/ContentServiceMain.java]: exclude DataSourceAutoConfiguration removed.

File: mongo-dao/src/main/resourses/application.properties (hunk 3 above)
Comment (Arundhathy-k) [reply mongo-dao/src/main/resourses/application.properties]: spring.config.import=aws-parameterstore:/config/NewsService/ and
spring.data.mongodb.uri= ${MongoDbURI} Removed.

File: newsService/src/main/resources/application.properties (hunk 4 above)
Comment (Arundhathy-k) [reply newsService/src/main/resources/application.properties:6]: Yes. We need this to connect application with monho db local database.

File: contentService/src/main/java/com/kovan/config/JpaConfig.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    @Value("${spring.jpa.properties.hibernate.format_sql}")
+    private boolean formatSql;
+
+    @Bean
+    public DataSource dataSource() {
+        return DataSourceBuilder.create()
+                .url(databaseUrl)
+                .username(username)
+                .password(password)
+                .driverClassName("com.mysql.cj.jdbc.Driver")
```
Comment (dhanyesh) [inline contentService/src/main/java/com/kovan/config/JpaConfig.java]: why only this driverClassName hardcoded

File: contentService/pom.xml [hunk 6]
```diff
@@ -43,11 +43,32 @@
...
+        <dependency>
+            <groupId>io.awspring.cloud</groupId>
+            <artifactId>spring-cloud-aws-starter-parameter-store</artifactId>
+            <version>3.2.1</version>
+        </dependency>
+
     </dependencies>
+
     <properties>
         <maven.compiler.source>20</maven.compiler.source>
```
Comment (dhanyesh) [inline contentService/pom.xml]: why do we have different value here some places 21 some places 20 can you stick with 21

File: newsService/src/main/java/com/kovan/app/service/NewsService.java [hunk 7]
```diff
@@ -51,23 +58,31 @@ public NewsService(RestTemplate restTemplate, ObjectMapper objectMapper,
         this.objectMapper = objectMapper;
         this.newsRepositoryService = newsRepositoryService;
     }
+
+    @Scheduled(cron = "0 0 */8 * * *")
```
Comment (dhanyesh) [inline newsService/src/main/java/com/kovan/app/service/NewsService.java]: do update branch based on latest develop

**Review by dhanyesh — CHANGES_REQUESTED** [review]

File: contentService/pom.xml (hunk 6 above)
Comment (Arundhathy-k) [reply contentService/pom.xml]: changed 20 to 21.

File: contentService/src/main/java/com/kovan/config/JpaConfig.java (hunk 5 above)
Comment (Arundhathy-k) [reply contentService/src/main/java/com/kovan/config/JpaConfig.java]: driver class name stored in parameter store and retrieved from there.

File: newsService/src/main/java/com/kovan/app/service/NewsService.java (hunk 7 above)
Comment (Arundhathy-k) [reply newsService/src/main/java/com/kovan/app/service/NewsService.java]: updated branch based on latest develop

**Pull request merged**